Helper/ProductSerializer.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Metrilo\Analytics\Helper;
+
+class ProductSerializer extends \Magento\Framework\App\Helper\AbstractHelper
+{
+    /**
+     * Product parent types
+     */
+    const PARENT_TYPES = [\Magento\ConfigurableProduct\Model\Product\Type\Configurable::TYPE_CODE, \Magento\Bundle\Model\Product\Type::TYPE_CODE, \Magento\GroupedProduct\Model\Product\Type\Grouped::TYPE_CODE];
+    
+    public function __construct(
+        \Magento\Store\Model\StoreManagerInterface                   $storeManager,
+        \Magento\Bundle\Model\Product\Type                           $bundleType,
+        \Magento\GroupedProduct\Model\Product\Type\Grouped           $groupedType,
+        \Magento\ConfigurableProduct\Model\Product\Type\Configurable $configurableType,
+        \Metrilo\Analytics\Model\ProductData                         $productData
+    ) {
+        $this->storeManager     = $storeManager;
+        $this->bundleType       = $bundleType;
+        $this->groupedType      = $groupedType;
+        $this->configurableType = $configurableType;
+        $this->productData      = $productData;
+    }
+    
+    public function serialize($product)
```

Comment:
Would you enjoy reading this code? Look at it and tell me some potential problems with it.

Reply:
You haven't answered this? :)